src/lib.rs

```diff
@@ -412,6 +424,162 @@ fn run_for_context<TContext: Sync>(
     })
 }
 
+pub fn run_for_slice_with_callback<'a>(
+    slice: impl Into<RopeOrSlice<'a>>,
+    tree: Option<&Tree>,
```

Comment:
Rather than eg a separate "entry point" for when you have the pre-parsed `tree_sitter::Tree`, just added it as an optional param to this existing one